bundles/org.openhab.binding.solax/src/main/java/org/openhab/binding/solax/internal/SolaxInverterHandler.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solax.internal;
+
+import java.time.ZonedDateTime;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.solax.internal.model.InverterData;
+import org.openhab.core.library.types.DateTimeType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolaxInverterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Konstantin Polihronov - Initial contribution
+ */
+@NonNullByDefault
+public class SolaxInverterHandler extends BaseThingHandler implements InverterDataUpdateListener {
+
+    private final Logger logger = LoggerFactory.getLogger(SolaxInverterHandler.class);
+
+    public SolaxInverterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    @Override
+    public void updateListener(InverterData data) {
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        logger.debug("Received a new inverter data object. Data = {}", data.toStringDetailed());
+        transferInverterDataToChannels(data);
+    }
+
+    private void transferInverterDataToChannels(InverterData data) {
+        // TODO check with reviewer if type and serial number is needed as a channel or as a property
+        updateProperty(SolaxBindingConstants.SERIAL_NUMBER, data.getWifiSerial());
```

Comment:
Perhaps the standard naming could be used:
```suggestion
        updateProperty(Thing.PROPERTY_SERIAL_NUMBER, data.getWifiSerial());
```